ct.sh

```diff
@@ -123,45 +49,17 @@ parse_command_line() {
     done
 }
 
-run_ct_container() {
-    echo 'Running ct container...'
-    local args=(run --rm --interactive --detach --network host --name ct "--volume=$(pwd):/workdir" "--workdir=/workdir")
+install_chart_testing() {
+    echo "Installing chart-testing..."
 
-    if [[ -n "$config" ]]; then
-        args+=("--volume=$(pwd)/$config:/etc/ct/ct.yaml" )
-    fi
-
-    args=("${args[@]}" "${docker_args[@]}")
-
-    args+=("$image" cat)
-
-    echo docker "${args[@]}"
-    docker "${args[@]}"
-    echo
-}
-
-configure_kube() {
-    # need to copy full .kube dir for certs, etc:
-    local confdir
-    confdir=$(dirname "$kubeconfig")
-    docker cp "$confdir" ct:/root/.kube
-}
-
-run_ct() {
-    echo "Running 'ct $command'..."
-    docker_exec ct "$command"
-    echo
-}
-
-cleanup() {
-    echo 'Removing ct container...'
-    docker kill ct > /dev/null 2>&1
-    echo 'Done!'
-}
+    curl -sSLo ct.tar.gz "https://github.com/helm/chart-testing/releases/download/$version/chart-testing_${version#v}_linux_amd64.tar.gz"
+    tar -xzf ct.tar.gz
+    sudo mv ct /usr/local/bin/ct
```

Comment:
This won't work so well with self-hosted runners.
How do you feel about using the APIs to cache tools and add them to the path, like done here: https://github.com/evryfs/cloudctl-action/blob/master/src/main.ts#L32

Reply:
Interesting. I guess on a self-hosted runner you could just pre-install the tools and not use the action at all. Anyways, I'll look into it. Installing to the cache dir would probably be better. For the configs that won't work yet, but I think we could introduce and environment variable for that.

Reply:
Updated. Depends on https://github.com/helm/chart-testing/pull/291

Reply:
Tested locally and here: https://github.com/unguiculus/gh-pages-helm-chart-repo-example/runs/1336625053?check_suite_focus=true
The jobs has to fail until the mentioned PR is merged.

Reply:
I still prefer installing through actions (in contrast to putting it into the runner image/machine) as this can let the runners be quite slim, and whatever the workflow requires is handled through the workflow. This gives more control to the end-user, and different versions of same tool can run on same runner.